.github/workflows/pull-request.yml

```diff
@@ -27,81 +29,81 @@ env:
 
 jobs:
 
-  supported-reqs:
+  build-basic:
 
-    name: 'Supported scripts/requirements.txt'
+    name: "build with requirements.txt"
     runs-on: ubuntu-20.04
 
     steps:
       - uses: actions/checkout@v2
 
-      - name: apt-get install
-        run: sudo apt-get -y install $ubuntu_base_deps
-
-      - name: PATH += .local/bin
-        run: echo "$HOME/.local/bin" >> $GITHUB_PATH
+      - name: setup env with requirements.txt
+        run: |
+          sudo apt-get -y install $ubuntu_base_deps
+          echo "$HOME/.local/bin" >> $GITHUB_PATH
+          pip install -r scripts/requirements.txt
 
-      - name: 'pip install -r scripts/requirements.txt'
-        run: pip install -r scripts/requirements.txt
+      - name: build doc
+        run: ./.github/actions/build-sof-doc.sh
 
-      # No YAML anchors; this is copy/paste
-      - name: configure SOF doc
+      - name: prepare file for deploy
+        if: ${{ github.event_name == 'push' && github.ref == 'refs/heads/publish' }}
         run: |
-          git clone https://github.com/thesofproject/sof
-          cmake -GNinja -S sof/doc -B _build_doxy
+          mkdir _build/latest
+          mv -v _build/html/* _build/latest
+          mv _build/latest _build/html/
+          touch _build/html/.nojekyll
+          mv scripts/publish-README.md _build/html/README.md
+          mv scripts/publish-index.html _build/html/index.html
+
+      - name: upload HTML for deploy
+        if: ${{ github.event_name == 'push' && github.ref == 'refs/heads/publish' }}
+        uses: actions/upload-artifact@v2
+        with:
+          name: html
+          path: _build/html
+
+  deploy:
+    needs: build-basic
+    runs-on: ubuntu-20.04
+    if: ${{ github.event_name == 'push' && github.ref == 'refs/heads/publish' }}
+    steps:
+      - name: download HTML
+        uses: actions/download-artifact@v2
```

Comment:
Can you add a comment explaining  which version gets downloaded and published? The last one that was just uploaded above I'm guessing?